pytransform3d/transform_manager/_temporal_transform_manager.py

```diff
@@ -44,14 +48,84 @@ class StaticTransform(TimeVaryingTransform):
     def __init__(self, A2B):
         self._A2B = A2B
 
-    def as_matrix(self, time):
+    def as_matrix(self, query_time):
         return self._A2B
 
     def check_transforms(self):
         self._A2B = check_transform(self._A2B)
         return self
 
 
+class NumpyTimeseriesTransform(TimeVaryingTransform):
+    """Transformation sequence, represented in a numpy array.
+    The interpolation is computed using screw linear interpolation (ScLERP)
+    method.
+
+    Parameters
+    ----------
+    time_arr: np.ndarray, shape (N,)
+        Numeric timesteps corresponding to the transformation samples.
+        You can use unixtime, relative time (starting with 0.0).
+    pq_arr : pandas.DataFrame, shape (N, 7)
+        Time-sequence of transformations, with each row representing a single
+        sample as position-quarternion (PQ) structure.
+    """
+
+    def __init__(self, time, pqs):
+
+        if len(pqs.shape) != 2:
+            raise ValueError("Shape of PQ array must be 2-dimensional.")
+
+        if time.size != pqs.shape[0]:
+            raise ValueError("Number of timesteps is not equal to number of PQ"
+                             "samples")
+
+        self._time_arr = time
+        self._pqs = pqs
+
+    def as_matrix(self, query_time):
+        pq = pq = self._interpolate_pq_using_sclerp(query_time)
+        return transform_from_pq(pq)
+
+    def _interpolate_pq_using_sclerp(self, query_time):
+
+        # identify the index of the preceding sample
+        t_arr = self._time_arr
+        idx_timestep_earlier_wrt_query_time = \
+            np.argmax(t_arr >= query_time) - 1
+
+        # deal with first timestamp
+        idx_timestep_earlier_wrt_query_time = max(
+            idx_timestep_earlier_wrt_query_time, 0
+        )
+
+        pq_array = self._pqs
+
+        # dual quaternion from preceding sample
+        t_prev = t_arr[idx_timestep_earlier_wrt_query_time]
+        pq_prev = pq_array[idx_timestep_earlier_wrt_query_time, :]
+        dq_prev = dual_quaternion_from_pq(pq_prev)
+
+        # dual quaternion from successive sample
+        t_next = t_arr[idx_timestep_earlier_wrt_query_time + 1]
+        pq_next = pq_array[idx_timestep_earlier_wrt_query_time + 1, :]
+        dq_next = dual_quaternion_from_pq(pq_next)
+
+        # since sclerp works with relative (0-1) positions
+        rel_delta_t = (query_time - t_prev) / (t_next - t_prev)
+        dq_interpolated = dual_quaternion_sclerp(dq_prev, dq_next, rel_delta_t)
+
+        return pq_from_dual_quaternion(dq_interpolated)
+
+    def check_transforms(self):
+        N = self._pqs.shape[0]
+        for i in range(N):
```

Comment:
You can use [pqs_from_transforms](https://github.com/dfki-ric/pytransform3d/blob/1f0b985bf799f616a2e493b6e43875e53c2ca229/pytransform3d/trajectories.py#L125)

Reply:
Reimplemented the whole method